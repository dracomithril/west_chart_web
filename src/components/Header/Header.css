.App-logo {
    height: 150px;
    margin-right: 10px;
}
.wcs_header {
    /*width: 1150px;*/
    height: 150px;
    color:white;
    background: #898e8e url(baner.png) no-repeat center;
    margin-bottom: 10px;
    min-width: 760px;
}
.title-header {
    margin: 10px 0 20px;
    border-bottom: 1px solid #eee;
    position: absolute;
    top: 60px;
    left: 20px;
}
.logged{
    /*height: 130px;*/
    color: white;
    float: right;
    max-width: 210px;
    text-align: left;
    padding: 10px 5px;
    margin-top: 10px;
    margin-bottom: 10px;
    margin-right: 5px;
    background-color: #898e8e;
    border: 1px solid white;
    border-radius: 4px;
    -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,.05);
    box-shadow: inset 0 1px 1px rgba(0,0,0,.05);
}
.logged img {
    width: 5rem;
    height: 5rem;
    margin-right: 1rem;
}
.logged div{
    padding: 2px 1px;
}
.logged button{
    padding: 2px 2px;
    margin: 2px;
    border: 1px solid white;
    width: 97%;
    border-radius: 5px;
}
.btn-spotify {
    color: #fff;
    background-color: forestgreen;
    border-color: rgba(0,0,0,0.2);
}
.btn-spotify:hover {
    color: #333;
    background-color: green;
    border-color: darkgreen;
}
.spotify{
    width: 97%;
    background: forestgreen;
    height: 30px;
    margin-top: 5px;
    text-align: center;
    border: 1px solid white;
    border-radius: 10px;
}
.spotify i{
    padding-right: 10px;
    padding-left: 5px;
    /*padding-top: 1px;*/
    font-size: 24px;
}
.spotify span{
    padding-top: 2px;
}