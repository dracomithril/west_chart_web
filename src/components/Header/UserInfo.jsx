/**
 * Created by Gryzli on 07.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Button, Image, Well } from 'react-bootstrap';
// TODO save information about user to database, do it somewhere else
// const { writeUserData } = require('./../../utils');

const UserInfo = ({ fb_user, sp_user, onLogoutClick }) => (
  <Well bsClass="logged">
    <div>
      <Image src={fb_user.picture_url} circle />
      <span>{`Hi, ${fb_user.first_name}`}</span>
      <br />
      <span>it is nice to see you again.</span>
      <br />
      {sp_user.id !== undefined && (
        <div className="spotify">
          <i className="fa fa-spotify" />
          <span>{sp_user.id}</span>
        </div>
      )}
      <div style={{ textAlign: 'center' }}>
        <Button
          bsStyle="warning"
          onClick={() => {
            onLogoutClick();
          }}
        >
          Sign out
        </Button>
      </div>
    </div>
  </Well>
);

UserInfo.propTypes = {
  fb_user: PropTypes.object,
  sp_user: PropTypes.object,
  onLogoutClick: PropTypes.func,
};

export default UserInfo;
