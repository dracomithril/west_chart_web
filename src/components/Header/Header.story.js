import React from 'react';
import { storiesOf } from '@storybook/react';
import { Provider } from 'react-redux';
import Header from './index';
import configureStore from '../../configureStore';

const store = configureStore({
  user: {
    picture_url: '/pic/concerned_deadpool.jpg',
    name: 'Deadpool',
    first_name: 'Wade',
  },
  sp_user: {
    id: 'unicorn',
  },
});

storiesOf('Header', module).add('Todo[VR]', () => (
  <Provider store={store}>
    <Header />
  </Provider>
));
