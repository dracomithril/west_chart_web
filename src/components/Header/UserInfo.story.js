import React from 'react';
import { storiesOf } from '@storybook/react';
import UserInfo from './UserInfo';

const spotifyUser = {
  id: 'deadpool',
};
const facebookUser = {
  picture_url: '/pic/concerned_deadpool.jpg',
  first_name: 'Deadpool',
};
storiesOf('UserInfo', module).add('Todo[VR]', () => (
  <UserInfo sp_user={spotifyUser} fb_user={facebookUser} />
));
