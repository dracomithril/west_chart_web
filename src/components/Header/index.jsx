/**
 * Created by Gryzli on 25.01.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { PageHeader } from 'react-bootstrap';
import CookieBanner from 'react-cookie-banner';
import UserInfo from './UserInfo';
import './Header.css';

import action_types from './../../reducers/action_types';

export default class Header extends React.Component {
  componentDidMount() {
    const { store } = this.context;
    this.unsubscribe = store.subscribe(() => this.forceUpdate());
  }

  componentWillUnmount() {
    this.unsubscribe();
  }

  onLogoutClick = () => {
    const { store } = this.context;
    store.dispatch({ type: action_types.SIGN_OUT_USER });
    sessionStorage.removeItem('fb_user');
    sessionStorage.removeItem('sp_user');
    window.location = '/';
  };

  render() {
    const { store } = this.context;
    const { user, sp_user } = store.getState();
    return (
      <div className="wcs_header">
        <CookieBanner
          message={
            "Yes, we use cookies. If you don't like it change website, we won't miss you! ;)"
          }
          onAccept={() => {}}
          cookie="user-has-accepted-cookies"
        />

        <PageHeader bsClass="title-header">Music Helper</PageHeader>
        {user.name && (
          <UserInfo fb_user={user} sp_user={sp_user} onLogoutClick={this.onLogoutClick} />
        )}
      </div>
    );
  }
}
Header.contextTypes = {
  store: PropTypes.object,
};
Header.propTypes = {};
