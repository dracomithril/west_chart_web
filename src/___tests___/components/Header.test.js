/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import Header from "./../../components/Header/index";
import Enzyme, {mount, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import {mountToJson, shallowToJson} from "enzyme-to-json";
import configureMockStore from "redux-mock-store";

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<Header/>', () => {
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = mount(
      <Header/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(mountToJson(wrapper)).toMatchSnapshot();
    let find = wrapper.find('UserInfo').getElements();
    expect(find).toHaveLength(0);
    wrapper.unmount();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
  it('render user info', () => {
    let state = Object.assign({}, initial_state, { user: { name: 'Simba' } });
    const store = mockStore(state);
    const wrapper = shallow(
      <Header/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    let find = wrapper.find('UserInfo').getElement();
    expect(find.type.name).toBe("UserInfo");
  });
});
