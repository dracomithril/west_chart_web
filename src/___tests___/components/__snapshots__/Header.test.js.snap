// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> render user info 1`] = `
<div
  className="wcs_header"
>
  <CookieBanner
    buttonMessage="Got it"
    cookie="user-has-accepted-cookies"
    cookieExpiration={
      Object {
        "years": 1,
      }
    }
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": true,
        "cookies": Object {},
        "hooks": undefined,
      }
    }
    dismissOnScroll={true}
    dismissOnScrollThreshold={0}
    message="Yes, we use cookies. If you don't like it change website, we won't miss you! ;)"
    onAccept={[Function]}
    styles={Object {}}
  />
  <PageHeader
    bsClass="title-header"
  >
    Music Helper
  </PageHeader>
  <UserInfo
    fb_user={
      Object {
        "name": "Simba",
      }
    }
    onLogoutClick={[Function]}
    sp_user={Object {}}
  />
</div>
`;

exports[`<Header/> renders without crashing ChartPresenter 1`] = `
<Header>
  <div
    className="wcs_header"
  >
    <CookieBanner
      buttonMessage="Got it"
      cookie="user-has-accepted-cookies"
      cookieExpiration={
        Object {
          "years": 1,
        }
      }
      cookies={
        Cookies {
          "HAS_DOCUMENT_COOKIE": true,
          "cookies": Object {},
          "hooks": undefined,
        }
      }
      dismissOnScroll={true}
      dismissOnScrollThreshold={0}
      message="Yes, we use cookies. If you don't like it change website, we won't miss you! ;)"
      onAccept={[Function]}
      styles={Object {}}
    >
      <BannerContent
        buttonMessage="Got it"
        message="Yes, we use cookies. If you don't like it change website, we won't miss you! ;)"
        onAccept={[Function]}
        styles={Object {}}
      >
        <div
          className="react-cookie-banner"
          style={
            Object {
              "backgroundColor": "#484848",
              "height": "45px",
              "position": "relative",
              "textAlign": "center",
              "width": "100%",
              "zIndex": "10000",
            }
          }
        >
          <span
            className="cookie-message"
            style={
              Object {
                "color": "#F0F0F0",
                "fontWeight": 500,
                "lineHeight": "45px",
              }
            }
          >
            Yes, we use cookies. If you don't like it change website, we won't miss you! ;)
          </span>
          <button
            className="button-close"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.6)",
                "border": "none",
                "borderRadius": "3px",
                "boxShadow": "none",
                "color": "#242424",
                "cursor": "pointer",
                "fontSize": "12px",
                "fontWeight": "500",
                "lineHeight": "24px",
                "marginTop": "-12px",
                "padding": "0 8px",
                "position": "absolute",
                "right": "35px",
                "top": "50%",
              }
            }
          >
            Got it
          </button>
        </div>
      </BannerContent>
    </CookieBanner>
    <PageHeader
      bsClass="title-header"
    >
      <div
        className="title-header"
      >
        <h1>
          Music Helper
        </h1>
      </div>
    </PageHeader>
  </div>
</Header>
`;

exports[`<Header/> renders without crashing ChartPresenter 2`] = `""`;
